### DISCUSSION TITLE
Gaining better control of split panes

### DISCUSSION THREAD

**Discussion by Lockszmith-GH** [zellij-org/zellij#2419] [Q&A]
This question is specifically about splitting on-the run. I'm ok with commands or setting up custom key-bindings if needed.

## Starting point

```plaintext
+------------------------+
|a                       |
|                        |
|                        |
+------------------------+
```

Then split-horiz:

```plaintext
+------------------------+
|a                       |
+------------------------+
|b                       |
+------------------------+
```

## My problem

Is there a command / key-binding that will produce the following vertical split:

```plaintext
12345678901234567890123456
+-----------+------------+
|a          |c           |
+-----------+            |
|b          |            |
+------------------------+
```

I know that if I verical-split first, I could get there, I'm asking:
What can I do without closing the panes I already have open?

**Comment by imsnif** [comment]
You might want to take a look at swap layouts: https://zellij.dev/documentation/swap-layouts.html

**Comment by Lockszmith-GH** [comment]
# Issue

Now  I see something else which I  might open an issue for later:

I see the following behavior:

## Starting point

Starting with (A):

```plaintext
+------------------------+
|a                       |
|                        |
|                        |
+------------------------+
```

Then split-right, I get (B):

```plaintext
+-----------+------------+
|a          |b           |
|           |            |
|           |            |
+------------------------+
```

Then split-down, makes (C):

```plaintext
+-----------+------------+
|a          |b           |
|           +------------+
|           |c           |
+------------------------+
```

Switch to `a` and split-down:, finally gets me (D):

```plaintext
+-----------+------------+
|a          |b           |
+-----------+------------+
|d          |c           |
+------------------------+
```

## Unexpected behavious

In pane `d` (as described above) when I close the pane I get:

```plaintext
+-----------+------------+
|a          |b           |
+-----------+------------+
|c                       |
+------------------------+
```

which I was not expecting, I was expecting to get the (C) state again.  
The same goes for killing `c`, `d` takes over the bottom half.

Changing the height of `d` (or `c`) produces the expected layout.

It seems that the order of creation of panes is not taken into account
when Zellij decides how to reconcile the geometry.

this might be by design, if that is the case, I would just like a
consistent way of splitting "within the pane" as well splitting "on the
outer bounds" of a group of panes (aka a tab).
